### PR TITLE
analysis: optimize performance and fix dead loop

### PR DESCRIPTION
**Pull request by howardjohn**
Two fixes:

1. We are infinite loop on stop when it triggers. `break` is for the
   switch, not the `for`.
2. No need to use regex
Fixes https://github.com/istio/istio/issues/40580

### CODE REVIEW

File: pkg/config/analysis/incluster/controller.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func (c *Controller) Run(stop <-chan struct{}) {
 			log.Debugf("finished enqueueing all statuses")
 		case <-stop:
 			t.Stop()
-			break
+			return
```
Comment (hzxuzhonghu) [inline pkg/config/analysis/incluster/controller.go:116]: This is a bug, but how is it related to CPU high?

**Review by hzxuzhonghu — APPROVED** [review]

File: pkg/config/analysis/incluster/controller.go (hunk 1 above)
Comment (howardjohn) [reply pkg/config/analysis/incluster/controller.go:116]: it's an infinite looping, end up with 100% cpu on selectgo. Once stop is closed

Comment (hzxuzhonghu) [reply pkg/config/analysis/incluster/controller.go:116]: I mean stop can be closed only when the process exit?

**Comment by howardjohn** [comment]
I think it's from losing leader election which is why the original issue
was only with multiple istiod replicas

**Pull request merged**